internal/util/temp.go

```diff
@@ -50,3 +50,29 @@ func pattern(obj client.Object) (p string) {
 	kind := strings.ToLower(obj.GetObjectKind().GroupVersionKind().Kind)
 	return fmt.Sprintf("%s-%s-%s-", kind, obj.GetNamespace(), obj.GetName())
 }
+
+// TODO: think of a better name?
+func WriteBytesToFile(bytes []byte, out string, temp bool) (*os.File, error) {
```

Comment:
Passing a path but not using it may be misleading.
Since the caller of this function is deleting the file, based on all the 3 usage that I've observed, it'd be better for the caller to create the file and pass `*os.File` to this function. Then you'd not need to return `*os.File`, the caller has a reference to delete the file.
For temporary file:
```go
f, _ := os.CreateTemp("", filepath.Base(out))
util.WriteBytesToFile(res.Bytes(), f)
defer os.Remove(f.Name())
```
For a given file path:
```go
f, _ := os.Create(path)
util.WriteBytesToFile(res.Bytes(), f)
defer os.Remove(f.Name())
```

Reply:
Maybe just `WriteToFile()` would be an intuitive name.